### PR TITLE
DOC: docs for `findExpansionElements()`

### PR DESCRIPTION
**Pull request by vlad20012**

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/macros/RsExpandedElement.kt [hunk 1]
```diff
@@ -95,6 +95,36 @@ private fun mapOffsetFromExpansionToCallBody(call: RsMacroCall, offset: Int): In
         ?.fromBodyRelativeOffset(call)
 }
 
+/**
+ * If [this] element is inside a macro call body and this macro is successfully expanded, returns
+ * a leaf element inside the macro expansion that is expanded from [this] element. Returns a
+ * list of elements because an element inside a macro call body can be placed in a macro expansion
+ * multiple times. Returns null if [this] element is not inside a macro call body, or the macro
+ * expanded erroneously.
```
Comment (Undin) [inline src/main/kotlin/org/rust/lang/core/macros/RsExpandedElement.kt]: Maybe "or the macro expansion failed" is better?

File: src/main/kotlin/org/rust/lang/core/macros/RsExpandedElement.kt [hunk 2]
```diff
@@ -95,6 +95,36 @@ private fun mapOffsetFromExpansionToCallBody(call: RsMacroCall, offset: Int): In
...
+ *
+ * # Examples
+ *
+ * Return an empty list if [this] element is not placed to an expansion:
```
Comment (Undin) [inline src/main/kotlin/org/rust/lang/core/macros/RsExpandedElement.kt]:
```suggestion
 * Returns an empty list if [this] element is not placed to an expansion:
```

File: src/main/kotlin/org/rust/lang/core/macros/RsExpandedElement.kt [hunk 3]
```diff
@@ -95,6 +95,36 @@ private fun mapOffsetFromExpansionToCallBody(call: RsMacroCall, offset: Int): In
...
+ *
+ * ```rust
+ * macro_rules foo { (bar) => { fn baz(){} } }
+ * foo!(bar); // This `bar` is matched with the `bar` in the pattern and is not placed to the expansion
+ * ```
+ *
+ * Return a list of single element if [this] element is placed into the expansion single time:
```
Comment (Undin) [inline src/main/kotlin/org/rust/lang/core/macros/RsExpandedElement.kt]:
```suggestion
 * Returns a single-element list if [this] element is placed into the expansion only once:
```

File: src/main/kotlin/org/rust/lang/core/macros/RsExpandedElement.kt [hunk 4]
```diff
@@ -95,6 +95,36 @@ private fun mapOffsetFromExpansionToCallBody(call: RsMacroCall, offset: Int): In
...
+ *
+ * ```rust
+ * macro_rules foo { ($i:ident) => { fn $i(){} } }
+ * foo!(bar); // This `bar` is placed to the expansion as a function name
+ * ```
+ *
+ * Return a list of multiple elements if [this] element is placed into the expansion multiple times:
```
Comment (Undin) [inline src/main/kotlin/org/rust/lang/core/macros/RsExpandedElement.kt]:
```suggestion
 * Returns a list of multiple elements if [this] element is placed into the expansion multiple times:
```

**Comment by vlad20012** [comment]
Fixed

**Pull request merged**